app/controllers/comment_controller.rb

```diff
@@ -99,12 +99,8 @@ def show_comments_for_user
   # Shows comments for a given object, most recent first. (Linked from the
   # "and more..." thingy at the bottom of truncated embedded comment lists.)
   def show_comments_for_target
-    model = begin
-              params[:type].to_s.constantize
-            rescue StandardError
-              nil
-            end
-    if !model || !model.acts_like?(:model)
+    model = safe_model_from_param_type
```

Comment:
NOTE: The new method performs slightly differently, and I think better than, the old one. In particular,
With params of a real, non-commentable model and real id:
- The old method would throw a runtime error, e.g., `Minitest::UnexpectedError:         RuntimeError: The model "APIKey" does not support comments!`
- The new method supplies a flash_error and returns gracefully.